content-services/content-sources-frontend
# HMS-4748: add warning for templates with expiring snaps

## Summary
This PR adds a warning for templates containing soon to be deleted snapshots, both in the template table and template detail.

![2024-12-05_10-09](https://github.com/user-attachments/assets/9ed3f1e6-78c6-422f-a583-6a4f3a1706fa)
![2024-12-05_10-09_1](https://github.com/user-attachments/assets/f01126da-b267-42ae-bf3c-115ede142796)

## Testing steps
1. Run the FE on local using a local backend. (`npm run local`)
2. Create a repo with two snapshots and a template that uses that repo with the date set back a year.
3. Set the date of the older one to the past in the DB. (Connect to the DB and execute this: `UPDATE snapshots SET created_at = (NOW() - CAST('360 days' AS INTERVAL)) WHERE uuid = '$SNAPSHOT_UUID';`)
4. Verify the template uses the old snapshot and the warnings are shown where they should.
